[AArch64][GlobalISel] Extend scalar lrint legalization.

davemgreen opened the pull request:
This extends the legalization of lrint, adding libcall support for fp128. The old vector legal types were removed as they were not being properly handled (vector lrint is a fairly new concept as far as I understand). They can be re-added properly in a followup.

arsenm commented on llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp:
```diff
@@ -1059,6 +1061,26 @@ LegalizerHelper::libcall(MachineInstr &MI, LostDebugLocObserver &LocObserver) {
+      return UnableToLegalize;
+    }
+    auto Libcall = getRTLibDesc(MI.getOpcode(), Size);
+    SmallVector<CallLowering::ArgInfo, 2> Args;
+    Args.push_back({MI.getOperand(1).getReg(), HLTy, 0});
```
Can do this in the initializer. Also don't need 2 size list for 1 element?

arsenm commented on llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp:
```diff
@@ -1059,6 +1061,26 @@ LegalizerHelper::libcall(MachineInstr &MI, LostDebugLocObserver &LocObserver) {
+    LegalizeResult Status =
+        createLibcall(MIRBuilder, Libcall, {MI.getOperand(0).getReg(), ITy, 0},
+                      Args, LocObserver, &MI);
+    if (Status != Legalized)
+      return Status;
+    break;
```
Erase the instruction and directly return? I don't know why the other cases here are mostly making you look to the end of the function to do the basic return

davemgreen replied:
SGTM, thanks. I don't mind changing the others too if you think it's a good idea.

davemgreen committed:
[AArch64][GlobalISel] Extend scalar lrint legalization.

This extends the legalization of lrint, adding libcall support for fp128. The
old vector legal types were removed as they were not being properly handled
(vector lrint is a fairly new concept as far as I understand). They can be
re-added properly in a followup.
